crates/router/src/core/payments/flows.rs

```diff
@@ -2759,3 +2763,115 @@ default_imp_for_post_session_tokens!(
+    setup_future_usage: Option<api_enums::FutureUsage>,
+    status: common_enums::AttemptStatus,
+) -> bool {
+    match status {
+        common_enums::AttemptStatus::Authorized => {
+            if let Some(api_enums::CaptureMethod::SequentialAutomatic) = capture_method {
+                match connector_name {
+                    router_types::Connector::Paybox => {
+                        // Check CIT conditions for Paybox
+                        (off_session == Some(true)
+                            || setup_future_usage == Some(api_enums::FutureUsage::OffSession))
+                            && customer_acceptance.is_some()
+                    }
+                    _ => false,
+                }
+            } else {
+                false
+            }
+        }
+        _ => false,
+    }
+}
+
+/// Executes a capture request by building a connector-specific request and deciding
+/// the appropriate flow to send it to the payment connector.
+pub async fn call_capture_request(
+    mut capture_router_data: types::RouterData<
+        api::Capture,
+        PaymentsCaptureData,
+        types::PaymentsResponseData,
+    >,
+    state: &SessionState,
+    connector: &api::ConnectorData,
+    call_connector_action: payments::CallConnectorAction,
+    business_profile: &domain::Profile,
+    header_payload: hyperswitch_domain_models::payments::HeaderPayload,
+) -> RouterResult<types::RouterData<api::Capture, PaymentsCaptureData, types::PaymentsResponseData>>
+{
+    // Build capture-specific connector request
+    let (connector_request, _) = capture_router_data
+        .build_flow_specific_connector_request(state, connector, call_connector_action.clone())
+        .await?;
+
+    // Execute capture flow
+    capture_router_data
+        .decide_flows(
+            state,
+            connector,
+            call_connector_action,
+            connector_request,
+            business_profile,
+            header_payload.clone(),
+        )
+        .await
+}
+
+/// Processes the response from the capture flow and determines the final status and the response.
+fn handle_post_capture_response(
+    authorize_router_data_response: types::PaymentsResponseData,
+    post_capture_router_data: Result<
+        types::RouterData<api::Capture, PaymentsCaptureData, types::PaymentsResponseData>,
+        error_stack::Report<ApiErrorResponse>,
+    >,
+) -> RouterResult<(common_enums::AttemptStatus, types::PaymentsResponseData)> {
+    match post_capture_router_data {
+        Err(err) => {
+            logger::error!(
+                "Capture flow encountered an error: {:?}. Proceeding without updating.",
+                err
+            );
+            Ok((
+                common_enums::AttemptStatus::Authorized,
+                authorize_router_data_response,
+            ))
+        }
+        Ok(post_capture_router_data) => {
+            match (
+                post_capture_router_data.response.clone(),
```

Comment:
why are we cloning here? we are only passing the reference right?

Reply:
not needed, will pass reference